extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php

```diff
@@ -48,10 +48,11 @@ public function Update() {
 		wfProfileIn( __METHOD__ );
 
 		$this->response->setVal( 'alwaysGrouped', empty( $wgEnableWallExt ) && empty( $wgEnableForumExt ) );
-		$this->response->setVal( 'notificationKey', $this->request->getVal( 'notificationKey' ) );
+		$this->response->setVal( 'notificationKey', htmlentities( $this->request->getVal( 'notificationKey' ), ENT_QUOTES ) );
```

Comment:
This is being output into a HTML attribute, so I'd prefer `Sanitizer::encodeAttribute`, and it's also generally preferable to escape as close to the output as possible, so I'd consider escaping in the template, but this is OK as well.